### PR TITLE
Make the header and announcement editable by CMS

### PR DESCRIPTION
**Pull request by ironng**
The nav and announcement banner now can be edited in the admin view. Refactors the header a bit to move the `Alert` into the header component from being a child of the `App`, so that we don't have to adjust the `top` positioning based on the content height of the announcement copy like we did previously.

### CODE REVIEW

File: src/header/nav.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+template: header-nav
+nav_items:
+  - label: Team
+    url: "#team"
+  - label: Advisors
+    url: "#advisors"
+  - label: Press
+    url: /press
+  - label: Blog
+    is_external_link: true
```
Comment (Shadowfiend) [inline src/header/nav.md:11]: FOR NOW :p

File: src/css/header.scss [hunk 2]
```diff
@@ -1,235 +1,229 @@
...
+                        }
 
-                        &:before {
-                            background-color: $link-color;
+                        &:focus,
+                        &:active,
+                        &.active.focus,
+                        &:active:focus,
+                        &.active,
+                        &.focus {
```
Comment (Shadowfiend) [inline src/css/header.scss:82]: Goodness gracious.

Realize this is a port-over from what we've already got, but I'll flag that I'm not sure we want necessarily focus and active to be identical. Something to look at more deeply later.

Comment (ironng) [reply src/css/header.scss:82]: Yes, good point!

File: gatsby-node.js [hunk 3]
```diff
@@ -5,9 +5,11 @@ const { fmImagesToRelative } = require("gatsby-remark-relative-images")
 exports.createPages = async ({ actions, graphql, reporter }) => {
   const { createPage } = actions
 
+  // Filter only for md files in the src/pages directory. Any md files found
+  // outside of that should not have a page created for it
   const result = await graphql(`
     {
-      allMarkdownRemark(limit: 1000) {
+      allMarkdownRemark(filter: { fileAbsolutePath: { regex: "/(pages)/" } }) {
```
Comment (Shadowfiend) [inline gatsby-node.js:12]: Not that it's a huge deal, and I don't know about performance here, but `glob: "*/pages/*"` might work here? Or is the capture group for `pages` significant?

File: src/components/Header.js [hunk 4]
```diff
@@ -4,64 +4,118 @@ import {
...
+    <header>
+      <Announcement />
+      <Navbar>
+        <Container fluid="md">
+          <NavScrollItem
+            element="div"
+            className="brand"
+            activeClass="active"
+            href="/"
+            to="home"
```
[the remark this answers is not in the document]
Comment (Shadowfiend) [reply src/components/Header.js:67]: Also I'm losing track of all the pieces, but what does `NavScrollItem` get us here vs `NavItem` or even a simple `Link`? This always goes to the site root, right?

[the remark this answers is not in the document]
Comment (ironng) [reply src/components/Header.js:67]: >Why both href and to here?

Derp. I don't remember. I wrote the original component quite some time ago. It _may_ have been to deal with some conflict between react-scroll and react-bootstrap at the time, but I'll clean it up later.

`NavScrollItem` is for items on the site root, and will auto scroll the page to that area. This made more sense with the original version of the site, which was designed to be a one-pager. `Link` is for any internal link and doesn't provide any scrolling.

File: src/components/Header.js [hunk 5]
```diff
@@ -4,64 +4,118 @@ import {
...
+          >
+            <Icons.Keep height="61px" width="235px" />
+          </NavScrollItem>
+          <NavbarToggler onClick={toggleNavbar} />
```
[the remark this answers is not in the document]
Comment (ironng) [reply src/components/Header.js:71]: I can add a class name like "mobile-only-toggle"?

File: gatsby-node.js (hunk 3 above)
Comment (ironng) [reply gatsby-node.js:12]: Will investigate!!

File: src/components/Header.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (Shadowfiend) [reply src/components/Header.js:71]: A class name would make more sense to me if we were using it to make decisions elsewhere, whereas mostly here I’m trying to make sure that a future dev isn’t going in circles trying to figure out where this weird button is when they don’t see it on their desktop browser—which a comment will do just fine. Up to you though :)

**Pull request merged**